bigchaindb_driver/pool.py

```diff
@@ -73,7 +123,9 @@ def get_connection(self):
             A :class:`~bigchaindb_driver.connection.Connection` instance.
 
         """
-        if len(self.connections) > 1:
-            return self.picker.pick(self.connections)
 
-        return self.connections[0]
+        if not self.nodes_available():
+            return None
+        elif len(self.connections) > 1:
+            return self.picker.pick(self.connections)
+        return self.connections[0]["node"]
```

Comment:
`pick` should deal with the special case of a single connection in the list. It makes sense to retry requests even when there is a single node configured.

Reply:
How should this be reconciled with the [exponential back-off](https://github.com/bigchaindb/bigchaindb-driver/pull/430#discussion_r194366351) i.e. In case of a single node should that back-off strategy be followed? 

- If no, then what should be the alternate approach?
- If yes, then following are the retry intervals (with `initial_delay = 1`): `2`, `4`, `8`, `16`. In this case should the client wait for all the attempts i.e. `16` seconds?

Reply:
Yes, it should follow the same strategy when there is only a single node.

The client should wait for as long as it wants, what can be expressed via an optional timeout parameter. Some more details [here](https://github.com/bigchaindb/bigchaindb-driver/pull/430/files#r194404305).